packages/wallet/backend/src/gatehub/client.ts

```diff
@@ -0,0 +1,239 @@
+import { createHmac } from 'crypto'
+import {
+  HTTP_METHODS,
+  ICreateManagedUserRequest,
+  ICreateManagedUserResponse,
+  ICreateTransactionRequest,
+  ICreateTransactionResponse,
+  IGetVaultsResponse,
+  ITokenRequest,
+  ITokenResponse
+} from '@/gatehub/types'
+import { Env } from '@/config/env'
+import {
+  DEFAULT_APP_SCOPE,
+  ONBOARDING_APP_SCOPE,
+  PAYMENT_TYPE,
+  PRODUCTION_CLIENT_IDS,
+  SANDBOX_CLIENT_IDS
+} from '@/gatehub/consts'
+import axios, { AxiosError } from 'axios'
+import { Logger } from 'winston'
+
+export class GateHubClient {
+  private clientIds = SANDBOX_CLIENT_IDS
+  private mainUrl = 'sandbox.gatehub.net'
+  constructor(
+    private env: Env,
+    private logger: Logger
+  ) {
+    if (this.env.NODE_ENV === 'production') {
+      this.clientIds = PRODUCTION_CLIENT_IDS
+      this.mainUrl = 'gatehub.net'
+    }
+  }
+
+  get apiUrl() {
+    return `https://api.${this.mainUrl}`
+  }
+
+  get rampUrl() {
+    return `https://managed-ramp.${this.mainUrl}`
+  }
+
+  get exchangeUrl() {
+    return `https://exchange.${this.mainUrl}`
+  }
+
+  get onboardingUrl() {
+    return `https://onboarding.${this.mainUrl}`
+  }
+
+  async getWithdrawalUrl(): Promise<string> {
+    const token = this.getIframeAuthorizationToken(
+      this.clientIds.onOffRamp,
+      DEFAULT_APP_SCOPE
+    )
+
+    return `${this.rampUrl}/?paymentType=${PAYMENT_TYPE.withdrawal}&bearer=${token}`
+  }
+
+  async getDepositUrl(): Promise<string> {
+    const token = this.getIframeAuthorizationToken(
+      this.clientIds.onOffRamp,
+      DEFAULT_APP_SCOPE
+    )
+
+    return `${this.rampUrl}/?paymentType=${PAYMENT_TYPE.deposit}&bearer=${token}`
+  }
```

Comment:
Should only have one method `getRampUrl(paymentType: PAYMENT_TYPE}` instead of two separate functions? The only difference is the `paymentType` query parameter.

Reply:
I will keep this way for now and changed if it makes sense on endpoint implementation